### ISSUE TITLE
Add support for ICMP

### ISSUE THREAD

**Issue by whysthatso** [TwinProduction/gatus#61]
that's a feature request. i am thinking about using gatus to replace https://github.com/arachnys/cabot but it's currently missing low level tests for infrastructure. any chance this would make sense to be implemented?

**Comment by TwinProduction** [comment]
Great idea @whysthatso.

I think something like this would be good:
```yaml
services:
  - name: example ICMP test
    url: "icmp://twinnation.org"
    interval: 30s
    conditions:
      - "[CONNECTED] == true"
      - "[IP] == 123.123.123.123"
      - "[RESPONSE_TIME] < 500"
```

I know the `icmp://` is totally unconventional, but I think it's more convenient to take that route for the sake of future proofing.

To clarify, TCP services are detected through the `tcp://` prefix while http services are supported through the `http://` or `https://` prefix, so it's not like it's _that_ unconventional (assuming we ignore the fact that `icmp://` isn't actually supported anywhere 😅)
The only exception to this rule is DNS, which just uses the IP of the DNS server.

I was thinking of adding support for UDP and GRPC in the future as well, which would both be prefixed with `udp://` and `grpc://` respectively, I just haven't gotten around to it yet.

**Comment by TwinProduction** [comment]
@whysthatso I've added support for monitoring services using ICMP. 
Could you give it a try and let me know if it works as expected? 

P.S. It's currently only in the `twinproduction/gatus:latest`.

**Comment by mharjac** [comment]
Hi!
I've tested it with this config:
```
- name: google.com
    url: "icmp://google.com"  
    interval: 30s  
    conditions:
      - "[CONNECTED] == true"
      - "[IP] == 216.58.212.142"
      - "[RESPONSE_TIME] < 5" 
```
and here is the result:
```
Timestamp:
2020-12-27 22:24:45
Response time:
0ms
Conditions:
X ~ [CONNECTED] == true (false == true)
✓ ~ [IP] == 216.58.214.238
✓ ~ [RESPONSE_TIME] < 1
Errors:
- Get "icmp://google.com": unsupported protocol scheme "icmp"
```
It always fails on `CONNECTED` condition. `IP` condition is working fine (it fails if I put some random IP), and it always passes on `RESPONSE_TIME`, although it shouldn't (my current ping is around 25 ms against this IP; it reports 0 ms  in web UI). Let me know if I can help you with anything else.

**Comment by TwinProduction** [comment]
@mharjac Thank you for getting back to me.

Are you sure that you re-pulled the latest version? 
I can replicate your issue, but only if I use an older version of Gatus that doesn't have support for ICMP.

Could you try it again after force re-pulling the image by using the following command:
```
docker pull twinproduction/gatus:latest
```

FYI, `[RESPONSE_TIME]` should always resolve to `0` if the connection has failed, because the response time calculated begins when the connection opens and it stops when the connection is closed. As such, if the connection fails, the response time should also be 0.

**Comment by mharjac** [comment]
Oh man... I didn't notice my "latest" image is actually 2 weeks old :D I even pulled repo and compiled it last night, and then for some reason, decided not to use it and go with docker container instead.
No errors this time! It works flawlessly! Thank you for this great feature and sorry for bothering you with this nonsense.

**Comment by TwinProduction** [comment]
@mharjac Hahaha don't worry about it, and I'm glad it's working as intended!

Released in [v1.10.0](https://github.com/TwinProduction/gatus/releases/tag/v1.10.0)

**Comment by whysthatso** [comment]
thanks so much, works great
